### DISCUSSION TITLE
Generic View with compiled bindings

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10789] [Q&A]
Is it possible to have a generic Window with compiled bindings?

So far so good...

```c#
public partial class AboutView<TSettings> : CommonWindow<AboutViewModelBase<TSettings>>
    where TSettings : SettingsDataBase, new() 
```

The view looks fine... until I want to set `x:DataType`

    x:DataType="app:AboutViewModelBase" x:TypeArguments="app:SettingsDataBase"

Nearly... 

It complains on AboutViewModelBase

> Incorrect number of type parameters

**Comment by timunie** [comment]
probably you need to set x:DataType to a non generic base class or an interface `ISettings`. That should work afaik.
